weather_dl/download_pipeline/pipeline.py

```diff
@@ -311,16 +258,24 @@ def run(argv: t.List[str], save_main_session: bool = True):
         pipeline_options.view_as(StandardOptions).runner = 'DirectRunner'
         manifest = LocalManifest(Location(local_dir))
 
-    pipeline_options = configure_workers(client_name, config, known_args.num_requests_per_key, pipeline_options)
+    num_requesters_per_key = known_args.num_requests_per_key
+    if num_requesters_per_key == -1:
+        num_requesters_per_key = CLIENTS[client_name](config).num_requests_per_key(
```

Comment:
You say you're making a guess here, which makes me think this guess being wrong might be a source of errors in the future. Perhaps log your guess to give your future self a way to debug?

Reply:
sgtm.